ethtxmanager/ethtxmanager.go

```diff
@@ -7,157 +7,283 @@ package ethtxmanager
+		err := c.state.SetSequenceGroupAsConfirmed(ctx, sequenceGroup.TxHash, nil)
+		if err != nil {
+			log.Errorf("failed to set sequence group as confirmed for batches %v tx %v: %v", sequenceGroup.BatchNumbers, sequenceGroup.TxHash.String(), err)
+			return false
+		}
+		log.Infof("sequence group for batches %v confirmed", sequenceGroup.BatchNumbers)
+		return true
+	}
+	log.Infof("sequence group for batches %v not confirmed yet", sequenceGroup.BatchNumbers)
+	return false
+}
+
+func (c *Client) tryReviewSequenceGroupTx(ctx context.Context, sequenceGroup state.SequenceGroup) {
+	// if it was not mined yet, check if the timeout since the last time the group was update has expired
+	lastTimeSequenceWasUpdated := sequenceGroup.CreatedAt
+	if sequenceGroup.UpdatedAt != nil {
+		lastTimeSequenceWasUpdated = *sequenceGroup.UpdatedAt
+	}
+	// if the time to review the tx has expired, we review it
+	if time.Since(lastTimeSequenceWasUpdated) >= c.cfg.IntervalToReviewSendBatchTx.Duration {
+		log.Infof("reviewing sequence group tx for batches %v due to long time waiting for it to be confirmed", sequenceGroup.BatchNumbers)
+
+		sequences, err := c.state.GetSequencesByBatchNums(ctx, sequenceGroup.BatchNumbers, nil)
 		if err != nil {
-			log.Errorf("failed to send batch verification, maximum attempts exceeded, err: %w", err)
-			return fmt.Errorf("failed to send batch verification, maximum attempts exceeded, err: %w", err)
+			log.Errorf("failed to get sequences by batch numbers: %v", err)
+			return
 		}
-		// Wait for tx to be mined
-		log.Infof("waiting for tx to be mined. Tx hash: %s, nonce: %d, gasPrice: %d", tx.Hash(), tx.Nonce(), tx.GasPrice().Int64())
-		err = c.ethMan.WaitTxToBeMined(ctx, tx, c.cfg.WaitTxToBeMined.Duration)
+
+		nonce := big.NewInt(0).SetUint64(sequenceGroup.TxNonce)
+		// using the same nonce, create a new transaction, this will make the gas to be
+		// recalculated with the current prices of the network
+		tx, err := c.ethMan.SequenceBatches(sequences, 0, nonce)
 		if err != nil {
-			if errors.Is(err, runtime.ErrOutOfGas) {
-				gas = increaseGasLimit(tx.Gas(), c.cfg.PercentageToIncreaseGasLimit)
-				log.Infof("out of gas with %d, retrying with %d", tx.Gas(), gas)
-				continue
-			} else if errors.Is(err, operations.ErrTimeoutReached) {
-				nonce = new(big.Int).SetUint64(tx.Nonce())
-				gasPrice = increaseGasPrice(tx.GasPrice(), c.cfg.PercentageToIncreaseGasPrice)
-				log.Infof("tx %s reached timeout, retrying with gas price = %d", tx.Hash(), gasPrice)
-				continue
+			// if the tx is already know, refresh the update date to give it more time to get mined
+			if errors.Is(err, core.ErrAlreadyKnown) {
+				err := c.state.UpdateSequenceGroupTx(ctx, sequenceGroup.TxHash, sequenceGroup.TxHash, nil)
+				if err != nil {
+					log.Errorf("give it more time to the sequence group related to the batches %v to get mined: %v", sequenceGroup.BatchNumbers, err)
+				}
```

Comment:
This sounds weird for me. If this error happens, it means that even the gasPrice of the resent tx is the same. In this case, wouldn't it be better to increase the gasPrice even more??

Reply:
exaplained above